tests/components/loqed/test_init.py

```diff
@@ -0,0 +1,106 @@
+"""Tests the init part of the Loqed integration."""
+
+import json
+from typing import Any
+from unittest.mock import AsyncMock, patch
+
+from loqedAPI import loqed
+
+from homeassistant.components.loqed.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_WEBHOOK_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry, load_fixture
+
+
+async def test_webhook_rejects_invalid_message(
+    hass: HomeAssistant,
+    hass_client_no_auth,
+    integration: MockConfigEntry,
+    lock: loqed.Lock,
+):
+    """Test webhook called with invalid message."""
+    await async_setup_component(hass, "http", {"http": {}})
+    client = await hass_client_no_auth()
+
+    coordinator = hass.data[DOMAIN][integration.entry_id]
+    lock.receiveWebhook = AsyncMock(return_value={"error": "invalid hash"})
+
+    with patch.object(coordinator, "async_set_updated_data") as mock:
```

Comment:
Don't patch the coordinator. That's an integration detail. Only assert some core state, eg an entity state in the state machine, or a patched library instance.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations